pyproj/geod.py

```diff
@@ -1067,3 +1115,49 @@ def __eq__(self, other: Any) -> bool:
             return False
 
         return self.__repr__() == other.__repr__()
+
+
+try:
+    import numpy as np  # type: ignore  # pylint: disable=import-outside-toplevel
+
+    def reverse_azimuth(azi: Any, radians: bool = False) -> Any:
```

Comment:
This function does not appear to behave the same as the one below and changes in place based on the existence of numpy. You should be able to re-write the function to work on a numpy array without needing numpy. I recommend using that approach as it is easier to test and ensure that the behavior is consistent with or without numpy.

Reply:
This may be useful:
- https://docs.python.org/3/library/math.html#math.copysign
- https://stackoverflow.com/a/1986776/11622836

Reply:
Yes, I thought about using `copysign`, but how would you apply this to a numpy array?
I couldn't find a way to write that function that will work on numpy array and on scalars without importing numpy.

Reply:
Some functions in the math module work on numpy arrays, but that one unfortunately doesn't. You will likely have to do a workaround for numpy arrays. Something like this:

```python
def sign(data):
    data_sign = data.copy()
    neg = data < 0
    data_sign[neg] = -1
    data_sign[~neg] = 1
    return data_sign
```

Reply:
This function would work on numpy array but not on scalars, so if you write two versions of the functions, why not use the built `sign` method as I did?

Reply:
It is an equivalent amount of code and easier to test/maintain if you don't use numpy.

Reply:
Actually, I have an idea that I think we both will like. Will share later when I am not on my phone.

Reply:
I see that numpy has also `copysign` so maybe now it's better.

Reply:
My main goal is to remove numpy from the core codebase. Otherwise, the tests and CI will need to be refactored to support both testing with and without numpy. The added complexity doesn't sound fun to maintain.

Here is an alternative approach: https://github.com/talos-gis/pyproj/pull/1

It uses cython instead of numpy and it supports all of the input scalar/array types that the geodesic routines support. This is easier to test and handles more input types in a performant way when numpy isn't present.

Reply:
Great, thanks for looking into this 👍